wandb/sdk/launch/agent/agent.py

```diff
@@ -734,3 +738,10 @@ def _check_run_finished(
             _logger.info("---")
             wandb._sentry.exception(e)
         return known_error
+
+    def get_job_and_queue(self) -> Optional[JobSpecAndQueue]:
+        for queue in self._queues:
+            job = self.pop_from_queue(queue)
+            if job is not None:
+                return JobSpecAndQueue(job, queue)
+        return None
```

Comment:
I think this actually is a regression from previous behavior. Before, it would prefer one queue but still cycle through all of them if the max_jobs is high enough. This implementation will totally ignore all queues except for the first as long as it has jobs available.

Something like `for queue in random.sample(self._queues, len(self._queues)):` would fix it. It would be non-deterministic but I don't think that's an issue?

Reply:
You're right, this does differ from the earlier implementation. While not ideal, I think this is an improvement over the issue we had where > max_jobs could be run.

I don't love the idea of adding randomness, the ideal solution is to have a round robin system. I think thats out of scope. Right now, no power users are running one agent on multiple queues (or we'd have issues with max_jobs) so this seems pretty benign as a regression of behavior.

I've created this ticket: https://wandb.atlassian.net/browse/WB-15692 however, this may not be done if we move the concurrency count to the queue